### PR TITLE
old pod deleter

### PR DESCRIPTION
**Pull request by grosser**
basic logic we need ... seems to work ...

@aksgithub

### CODE REVIEW

**Comment by ankilosaurus** [comment]
what's the use case for this ?

**Comment by grosser** [comment]
"make kube-remediator cleanup old console pods"
these pods need to die if they do not get cleaned up .. also could be good for apps that have memory leaks :D

**Comment by ankilosaurus** [comment]
> "make kube-remediator cleanup old console pods"
> these pods need to die if they do not get cleaned up .. also could be good for apps that have memory leaks :D

For memory leaks, we should just let it OOMkilled and taken care of by Crashloop so it does not go under the radar.

File: pkg/remediator/oldpoddeleter.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	pods, err := p.client.GetPods("", metav1.ListOptions{
+		LabelSelector: "kube-remediator/OldPodDeleter=true",
+	})
+	if err != nil {
+		p.logger.Error("Error getting pod list: ", zap.Error(err))
+		return
+	}
+
+	// deleteOldPods those that are too old
+	cutoff := time.Now().Add(-24 * time.Hour)
```
Comment (ankilosaurus) [inline pkg/remediator/oldpoddeleter.go:51]: what if definition of "old" varies among applications? Debugging session more than 24 hours might be unrealistic but just a thought.

**Comment by grosser** [comment]
the console pods are supposed to auto-kill after 24h so that's not a
problem ... we can do varian amounts of age with an annotation, but so far
we only have 1 usecase, so 24h can be the default :)

File: pkg/remediator/oldpoddeleter.go (hunk 1 above)
Comment (ankilosaurus) [reply pkg/remediator/oldpoddeleter.go:51]: how about read cutoff from application pod's annotation?

**Review by ankilosaurus — APPROVED** [review]

**Comment by grosser** [comment]
for now as simple as possible, the only things I want to get done is the
reuse/testing ... rest can wait :)
... we can pair on that tomorrow

**Pull request merged**